staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -296,6 +307,29 @@ func (f *featureGate) Add(features map[Feature]FeatureSpec) error {
 	return nil
 }
 
+func (f *featureGate) OverrideDefault(name Feature, override bool) error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	known := map[Feature]FeatureSpec{}
+	for name, spec := range f.known.Load().(map[Feature]FeatureSpec) {
+		known[name] = spec
+	}
+
+	if spec, ok := known[name]; !ok {
+		return fmt.Errorf("cannot override default: feature %q is not registered", name)
+	} else if spec.LockToDefault && override != spec.Default {
+		return fmt.Errorf("cannot override default: feature %q default is locked to %t", name, spec.Default)
+	} else {
+		spec.Default = override
```

Comment:
If we prevent overwriting when this flag has already been added to a FlagSet, the override will be reported on the command line, and I like it.(https://github.com/kubernetes/kubernetes/blob/e95179ffb6ee236aa3ff55c931d2078713d3a70e/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L330).